clients/python-wrapper/lakefs/reference.py

```diff
@@ -192,3 +176,28 @@ def __str__(self) -> str:
 
     def __repr__(self):
         return f"lakefs://{self._repo_id}/{self._id}"
+
+
+def get_generator(func, *args, max_amount: Optional[int] = None, **kwargs):
+    """
+    Generic generator function, for lakefs-sdk listings functionality
+
+    :param func: The listing function
+    :param args: The function args
+    :param max_amount: The max amount of objects to generate
+    :param kwargs: The function kwargs
+    :return: A generator based on the listing function
+    """
+    has_more = True
+    with api_exception_handler():
+        while has_more:
+            page = func(*args, **kwargs)
+            has_more = page.pagination.has_more
+            kwargs["after"] = page.pagination.next_offset
+            for res in page.results:
+                yield res
+
+                if max_amount is not None:
+                    max_amount -= 1
+                    if max_amount <= 0:
```

Comment:
looks like we are still not passing any value to the server to manage the max amount required for listing - can be in a different PR.

Reply:
We are passing any additional value via the kwargs - there's no need for any other param

Reply:
It is back to the fact that we provide some collection interface and passing max_amount and using it only on client side to limit the iteration, which I think is not required.
We have a different issue for max_amount we can resolve it not in this pr.